.github/workflows/test.yml

```diff
@@ -92,6 +96,87 @@ jobs:
         uses: actions/cache@v2
         with:
           path: .tox
-          key: tox-cache-${{ matrix.tox-environment }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}
+          key: tox-cache-${{ matrix.tox-environment }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
+      - name: run tox
+        run: tox -e ${{ matrix.tox-environment }}
+  contrib-build:
+    env:
+      # We use these variables to convert between tox and GHA version literals
+      py35: 3.5
+      py36: 3.6
+      py37: 3.7
+      py38: 3.8
+      pypy3: pypy3
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false  # ensures the entire test matrix is run, even if one permutation fails
+      matrix:
+        python-version: [ py35, py36, py37, py38, pypy3 ]
+        package: ["instrumentation", "exporter"]
+        os: [ ubuntu-latest ]
+        include:
+          # py35-instrumentation segfaults on 18.04 so we instead run on 20.04
+          - python-version: py35
+            package: instrumentation
+            os: ubuntu-20.04
+        exclude:
+          - os: ubuntu-latest
+            python-version: py35
+            package: instrumentation
+    steps:
+      - name: Checkout Contrib Repo @ SHA - ${{ env.CONTRIB_REPO_SHA }}
+        uses: actions/checkout@v2
+        with:
+          repository: open-telemetry/opentelemetry-python-contrib
+          ref: ${{ env.CONTRIB_REPO_SHA }}
+      - name: Checkout Core Repo @ SHA ${{ github.sha }}
+        uses: actions/checkout@v2
+        with:
+          repository: open-telemetry/opentelemetry-python
+          path: opentelemetry-python-core
```

Comment:
They are nested, so it's flipped, and in this case the Contrib Repo is the working directory and the Core Repo is nested.

For the step above this when it checks out the Contrib repo, you'll notice it doesn't have the `path` parameter. This means checkout all the files at the root.

For the step when it checks out the Core repo, it checkouts all the files under the `opentelemetry-python-core/` folder because that is what is specified under `path`.

The [docs on the Checkout Action] I found have this to say:
```
# Relative path under $GITHUB_WORKSPACE to place the repository
    path: ''
```

Since this is a new job, `contrib-build`, it's a clean slate where the Contrib repo is the root.